### PR TITLE
Duplicate of images in templates

### PR DESCRIPTION
**Pull request by 0xecute**
Add feature to have several times the same picture in the template.
Add test.
Add requirements.txt
Update README

### CODE REVIEW

**Comment by 0xecute** [comment]
@werdeil  @anxuae  Do you intend to merge this soon?

File: pibooth_picture_template.py [hunk 1]
```diff
@@ -111,13 +111,11 @@ def parse(self):
             orientation = pictures.PORTRAIT if size[0] < size[1] else pictures.LANDSCAPE
 
             captures = self.parse_captures(template)
-            subdata = data.setdefault(orientation, {}).setdefault(len(captures), {})
-            if subdata:
-                raise TemplateParserError("Several templates with {} captures are defined".format(
-                    len(captures)))
```
Comment (werdeil) [inline pibooth_picture_template.py:117]: why did you remove this part? this check is useful

File: tests/test_parser.py [hunk 2]
```diff
@@ -49,3 +52,4 @@ def test_text_positions(nbr, template):
     for rect in parser.get_text_rects(nbr):
         rect = pygame.Rect(*rect[:4])
         assert rect.colliderect(pygame.Rect(0, 0, *parser.get_size(nbr)))
+
```
Comment (werdeil) [inline tests/test_parser.py]: Leave only one empty line at the end of a file

File: tests/test_parser.py [hunk 3]
```diff
@@ -22,6 +22,9 @@ def test_capture_text(nbr, template):
             parser.get_capture_rects(nbr)
         with pytest.raises(TemplateParserError):
             parser.get_text_rects(nbr)
+    elif template.endswith('symetric_template.xml'):
+        assert len(parser.get_capture_rects(nbr)) == nbr * 2 # Excepecting 2 times more pictures to be captured
+        assert len(parser.get_text_rects(nbr)) == 2 * 2 # Excepecting 2 times more texts to be captured
```
Comment (werdeil) [inline tests/test_parser.py]: Typo
```suggestion
        assert len(parser.get_capture_rects(nbr)) == nbr * 2 # Excpecting 2 times more pictures to be captured
        assert len(parser.get_text_rects(nbr)) == 2 * 2 # Excpecting 2 times more texts to be captured
```

File: pibooth_picture_template.py [hunk 4]
```diff
@@ -395,13 +403,13 @@ def _build_outlines(self, image):
...
         draw = ImageDraw.Draw(image)
-        for pos_x, pos_y, width, height, angle in self._iter_images_rects():
+        for pos_x, pos_y, width, height, angle,value in self._iter_images_rects():
             rect = Image.new('RGBA', (width, height), (255, 0, 0, 0))
             draw = ImageDraw.Draw(rect)
             draw.rectangle(((0, 0), (width - 1, height - 1)), outline='red')
             self._image_paste(rect, image, pos_x, pos_y, angle)
         if self._texts:
-            for pos_x, pos_y, width, height, angle in self._iter_texts_rects():
+            for pos_x, pos_y, width, height, angle,value in self._iter_texts_rects():
```
Comment (werdeil) [inline pibooth_picture_template.py]: I don't understand why did you add the value if you don't use it?

File: pibooth_picture_template.py [hunk 5]
```diff
@@ -395,13 +403,13 @@ def _build_outlines(self, image):
         :type image: :py:class:`PIL.Image`
         """
         draw = ImageDraw.Draw(image)
-        for pos_x, pos_y, width, height, angle in self._iter_images_rects():
+        for pos_x, pos_y, width, height, angle,value in self._iter_images_rects():
```
Comment (werdeil) [inline pibooth_picture_template.py]: I don't understand why did you add the value if you don't use it?

File: pibooth_picture_template.py [hunk 6]
```diff
@@ -199,9 +199,6 @@ def parse_captures(self, mxgraph_node):
         """
         captures = [cell for cell in mxgraph_node.iter('mxCell')
                     if cell.get('vertex') == "1" and not cell.get('style').startswith('text;')]
-        if len(captures) > 4 or len(captures) < 1:
-            raise TemplateParserError("Too many captures positions ({}) in template '{}' (1 to 4 expected)".format(
-                len(captures), mxgraph_node.get('name')))
```
Comment (werdeil) [inline pibooth_picture_template.py:204]: I don't understand why you need to remove this protection

**Review by werdeil — CHANGES_REQUESTED** [review]
Here are some comments on the PR, I didn't test it so I won't be able to give a full feedback, but at least you have some changes to do and some questions to answer.

File: pibooth_picture_template.py (hunk 6 above)
Comment (0xecute) [reply pibooth_picture_template.py:204]: Because we can expect unlimited number of captures, as you can duplicate them as much as you want.

File: pibooth_picture_template.py (hunk 4 above)
Comment (0xecute) [reply pibooth_picture_template.py]: I use it above ( line 370), here I added it because otherwise it throws an error.

File: pibooth_picture_template.py (hunk 5 above)
Comment (0xecute) [reply pibooth_picture_template.py]: I use it above ( line 354), here I added it because otherwise it throws an error.

File: pibooth_picture_template.py (hunk 1 above)
Comment (0xecute) [reply pibooth_picture_template.py:117]: You are right, I should keep a check for this

File: pibooth_picture_template.py [hunk 7]
```diff
@@ -128,8 +126,10 @@ def parse(self):
                     LOGGER.warning("Template capture '%s' Y-position out of bounds, try to auto-adjust",
                                    capture.get('value'))
                     posy = posy % size[1]
-                subdata.setdefault('captures', []).append((posx, posy, width, height, rotation))
+                _captures.append((posx, posy, width, height, rotation,capture.get('value')))
```
Comment (anxuae) [inline pibooth_picture_template.py]: to be remove "_", this is not a private variable

File: requirements.txt [hunk 8]
```diff
@@ -0,0 +1,4 @@
+pibooth==2.0.5
```
Comment (anxuae) [inline requirements.txt]: requirements.txt to be removed
Already handled in the setup.py

File: README.rst [hunk 9]
```diff
@@ -275,3 +289,12 @@ Step 6: save the template file
...
    :width: 80 %
    :alt: step6_2_export
+
+Development
+-------------------
+
+
+::
+
+    $ pip3 install -r requirements.txt
```
Comment (anxuae) [inline README.rst]: No, the way to do it in pibooth project is use the setup.py:

pip install -e .

File: pibooth_picture_template.py [hunk 10]
```diff
@@ -111,13 +111,11 @@ def parse(self):
...
-            subdata = data.setdefault(orientation, {}).setdefault(len(captures), {})
-            if subdata:
-                raise TemplateParserError("Several templates with {} captures are defined".format(
-                    len(captures)))
-
+            _captures = []
+            count_captures = set()
             for capture in captures:
                 style = self.parse_style(capture)
+                count_captures.add(capture.get('value'))
```
Comment (anxuae) [inline pibooth_picture_template.py]: This is not retro compatible with previous version of the plugin where numbering was not mandatory
